orion/pkg/domains/regex.go

```diff
@@ -0,0 +1,22 @@
+package domains
+
+const (
+	// Alphanumeric characters separated by underscores
+	REGEX_PROGRAM_ID = `^[[:alnum:]]+(_[[:alnum:]]+)*$`
+
+	// In the form "year_season" i.e. 2020_fall
+	REGEX_SEMESTER_ID = `^[1-9]\d{3,}_((spring)|(summer)|(fall)|(winter))$`
+
+	/* Starts with a capital letter or number. Words consist of alphanumeric characters and dashes, spaces, and
+	underscores separate words. Words can have parentheses around them and number signs must be followed by numbers. */
+	REGEX_TITLE = `^[A-Z0-9][[:alnum:]-]*([- _]([(]?#\d[)]?|&|([(]?[[:alnum:]]+[)]?)))*$`
```

Comment:
nit: (optional change) It would be great to make this more descriptive as well but I don't have a good suggestion atm.
Ideally, when I use `REGEX_TITLE`, I should have a good idea of what it does just from reading the variable name. This helps us eliminate comments from the code or having to go back and forth to figure out what variables / functions do what.